WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductPricingFragment.kt

```diff
@@ -337,16 +373,19 @@ class ProductPricingFragment : BaseProductFragment(), ProductInventorySelectorDi
                     val selectedProductTaxClass = viewModel.getTaxClassBySlug(selectedTaxClass)
                     selectedProductTaxClass?.let {
                         product_tax_class.setText(it.name)
-                        viewModel.updateProductDraft(taxClass = it.slug)
+                        viewModel.onDataChanged(taxClass = it.slug)
                     }
                 }
             }
         }
-
-        changesMade()
     }
 
     override fun onRequestAllowBackPress(): Boolean {
-        return viewModel.onBackButtonClicked(ExitPricing())
+        return if (viewModel.event.value == Exit) {
+            true
+        } else {
+            viewModel.onExit()
+            false
+        }
```

Comment:
np: just a suggestion, nothing essential, but if you want, you can take advantage of the generated boolean of the comparison itself and hide the explicit Boolean returns and use the `also` for the `false` scenario where the `onExit()` needs to be called.

```suggestion
        return (viewModel.event.value == Exit)
        .also { if(it.not()) viewModel.onExit() }
```

Reply:
Thanks 👍